### ISSUE TITLE
dev-apprenticeship: README promises auto-confidence growth and knowledge decay, but no agent ever updates confidence

### ISSUE THREAD

**Issue by ylohnitram** [Replikanti/agentis-colonies#100]
## Problem

`dev-apprenticeship/README.md` claims:

> **After promotion**: Suggestions appear in logs (0.6) or directly on GitLab (0.85). Knowledge grows with every tick. **Agents that predict correctly gain confidence. Stale knowledge decays.** The system improves as long as you keep working on the project.

Neither clause on the bolded sentence is true:

1. **Agents don't gain confidence.** Confidence is purely read (`get_confidence()`), never written, by any of the 21 agents:

   ```bash
   $ grep -rn -E 'memo_write.*confidence|get_confidence\(\)' dev-apprenticeship/*/agents/*.ag | grep memo_write
   # no matches
   ```

   Every agent reads `get_confidence()` to decide what to do. No agent ever calls `memo_write("<name>:confidence", ...)`. The only way confidence ever changes is when the operator runs `agentis memo set`.

2. **No knowledge decay logic** is visible in the colonies repo. Each `learn()` call adds an entry; nothing prunes or down-weights stale entries. (May or may not be handled runtime-side in agentis-core — worth confirming, but the README frames it as a dev-apprenticeship property.)

## Impact

This breaks the core value proposition. The README sells a model where you start agents at 0.5, keep working, and they **earn their way up** based on how well they predict your behavior. In reality:

- Day 1 at 0.5: agents observe.
- Day 60 at 0.5: agents still observe. Forever.
- The only way anything autonomous ever happens is you manually run `agentis memo set <agent>:confidence 0.85` — which defeats the "apprenticeship" metaphor entirely. You're not watching them grow; you're just deciding when to hand over the keys.

Operators following the README will keep checking the dashboard expecting the confidence bars to climb, and will eventually realize nothing is happening without their manual intervention. Same trust-erosion pattern as #99.

## Two ways to resolve

**Option A — Rewrite the promotion section to match reality.**
Reframe the confidence gradient as "operator-controlled": you decide when each agent is ready based on what you see in logs / the suggestion feed. Drop the "agents earn confidence" narrative. 30-minute doc edit.

**Option B — Implement auto-confidence.**

Needs a feedback signal. Sketch:

1. When an agent emits a suggestion at 0.6-0.84 (e.g. `labeler` suggests label X for issue #42), record the suggestion with the issue ID.
2. On the next tick, check what the *human* actually did to that issue (did they apply label X, a different label, or none?).
3. Match → `confidence += learning_rate`. Mismatch → `confidence -= learning_rate`. No action yet → wait.
4. Similarly for decay: penalize knowledge entries that never recurred after N ticks.

This is meaningful work — per-agent feedback signal, persistence of "pending verdict" suggestions, tick-over-tick matching logic — but it's what makes the apprenticeship model actually apprenticeship.

## Suggested scope

Option A unblocks honesty immediately and stops misleading new users. Option B is the real product feature and probably deserves its own issue/MR per colony (the feedback signal for `labeler` is different from `code_writer` is different from `approval_decider`).

If the project owner wants the apprenticeship framing to survive, Option B has to happen eventually — otherwise the whole "silent → suggest → act" gradient is just UI around a manual kill switch.

## Related

- #99 (README advertises `personal`/`project:` knowledge tags that don't exist in implementation)

Same pattern: README describes a feature that isn't in the code.

## Environment

- agentis-colonies: master
- agentis: v1.1.7

**Comment by ylohnitram** [comment]
## Decision: Option A first, then Option B per colony

Going with **Option A now** (rewrite the promotion section to match current operator-controlled reality) and treating **Option B as follow-up work, split per colony** in separate MRs.

Rationale:

- Option A is a 30-minute doc edit that immediately stops the "I'm waiting for my agents to grow up" trap. Current README is actively misleading.
- Option B is meaningful product work. The feedback signal is different for each colony (`labeler` vs `code_writer` vs `approval_decider` vs `release_checker` — each has a different "ground truth" to compare against). Doing all five under one MR would be a mess and rush the design.
- Suggested split: one design ADR / issue per colony that proposes its feedback signal, then implementation MRs per colony. Start with `labeler` (simplest ground truth: suggested label vs. applied label) to validate the pattern.
- Same decision as #99: doc honesty first, feature later.

README edit will land first. Per-colony feature work follows when there's bandwidth.
